Preserve distributionSha256Sum in gradle-wrapper.properties by Wrappe… (#22220)
gradle/gradle: subprojects/docs/src/docs/userguide/reference/gradle_wrapper.adoc

Gradle will report a build failure in case the configured checksum does not match the checksum found on the server for hosting the distribution. Checksum Verification is only performed if the configured Wrapper distribution hasn't been downloaded yet.

[NOTE]
====

> Executing Wrapper task will preserve `distributionSha256Sum` in `gradle-wrapper.properties` if it was declared.

Reviewer: ```suggestion
The `Wrapper` task will fail if the `gradle-wrapper.properties` contains `distributionSha256Sum` and the task configuration does not define. Also, executing the `Wrapper` task will preserve the `distributionSha256Sum` configuration if the Gradle version is unchanged.
```

> The `Wrapper` task will fail if the `gradle-wrapper.properties` contains `distributionSha256Sum` and the task configuration does not define. Also, executing the `Wrapper` task will preserve the `distributionSha256Sum` configuration if the Gradle version is unchanged.

Reviewer 2: ```suggestion
The `Wrapper` task fails if `gradle-wrapper.properties` contains
`distributionSha256Sum`, but the task configuration does not define a sum.
Executing the `Wrapper` task preserves the `distributionSha256Sum`
configuration when the Gradle version does not change.
```
Suggestion: minor wording improvements.
